stagemonitor-core/src/main/java/org/stagemonitor/core/CorePlugin.java

```diff
@@ -684,7 +703,36 @@ public URL getElasticsearchUrl() {
 			return null;
 		}
 		final int index = accessesToElasticsearchUrl.getAndIncrement() % urls.size();
-		return urls.get(index);
+		URL elasticsearchURL = urls.get(index);
+
+		final String defaultUsernameValue = elasticsearchDefaultUsername.getValue();
+		final String defaultPasswordValue = elasticsearchDefaultPassword.getValue();
+
+		if (elasticsearchURL.getUserInfo() == null
+				&& ! defaultUsernameValue.isEmpty()
+				&& ! defaultPasswordValue.isEmpty()) {
+
+			try {
+				String username = URLEncoder.encode(defaultUsernameValue, "UTF-8");
+				String password = URLEncoder.encode(defaultPasswordValue, "UTF-8");
+				StringBuilder stringBuilder = new StringBuilder();
+				stringBuilder
+						.append(elasticsearchURL.getProtocol())
+						.append("://")
+						.append(username)
+						.append(":")
+						.append(password)
+						.append("@")
+						.append(elasticsearchURL.getHost())
+						.append(":")
+						.append(elasticsearchURL.getPort());
```

Comment:
Looks good to me. I found one possible problem (and I do not think that this happens often). If someone uses Elasticsearch behind a reverse-proxy and configures it to pass requests from `server1.example-internal.com/search` to `some-elasticsearch-host.example-internal.com`, then we would need to append `elasticsearchURL.getPath()` as well, so that stagemonitor sends the request to `server1.example-internal.com/search` instead of `server1.example-internal.com`.